pkg/services/ngalert/schedule/schedule.go

```diff
@@ -653,7 +637,43 @@ func (r *alertRuleRegistry) keyMap() map[models.AlertRuleKey]struct{} {
 type alertRuleInfo struct {
 	evalCh  chan *evalContext
 	stopCh  chan struct{}
-	version int64
+	mtx     sync.Mutex
+	stopped *atomic.Bool // indicates that the stopCh was closed and therefore the rule evaluation routine should exit
```

Comment:
I don't understand the need for `*atomic.Bool` when we also lock `mtx`? Can't this just be a `bool`?

Reply:
The reason I used atomic boolean was that there is a guess that using a regular bool can make race detector complain about it.

Reply:
I applied your suggestion and there is no need in the atomic bool anymore. So I replaced it with just bool